satellite/contact/endpoint.go

```diff
@@ -28,10 +32,67 @@ func NewEndpoint(log *zap.Logger, service *Service) *Endpoint {
 // Checkin is periodically called by storage nodes to keep the satellite informed of its existence,
 // address, and operator information. In return, this satellite keeps the node informed of its
 // reachability.
+// When a node checkins with the satellite, the satellite pings the node back to confirm they can
+// successfully connect.
 func (endpoint *Endpoint) Checkin(ctx context.Context, req *pb.CheckinRequest) (_ *pb.CheckinResponse, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	// TODO: record information, pingback node here
+	pingNodeSuccess, pingErrorMessage, err := pingBack(ctx, endpoint, req)
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+
+	// Update the overlay cache with new uptime info from pinging back the node
+	peerID, err := peerIDFromContext(ctx)
+	_, err = endpoint.service.overlay.UpdateUptime(ctx, peerID, pingNodeSuccess)
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+
+	// Should we update the req.NodeCapacity here as well?
+
+	return &pb.CheckinResponse{
+		PingNodeSuccess:  pingNodeSuccess,
+		PingErrorMessage: pingErrorMessage,
+	}, nil
+}
+
+func pingBack(ctx context.Context, endpoint *Endpoint, req *pb.CheckinRequest) (bool, string, error) {
+	client, err := newClient(ctx,
+		endpoint.log,
+		endpoint.service.transport,
+		req.GetAddress(),
+	)
+	defer func() {
```

Comment:
I believe this `defer` should come after the following err check. If `err != nil` and we return on line 70, we don't need to close anything because we shouldn't have an open stream.